spec/unittest.dd

```diff
@@ -8,87 +8,113 @@ $(GRAMMAR
 )
+    $(P Unit tests are a builtin framework of test cases
+    applied to a module to determine if it is working properly.
+    A D program can be run with unit tests enabled or disabled.
+    )
 
-        $(P Unit tests are a series of test cases applied to a module to determine
-        if it is working properly. Ideally, unit tests should be run every
-        time a program is compiled.
-        )
+    $(P Unit tests are a special function defined like:)
 
-        $(P Unit tests are a special function defined like:)
+    ---
+    unittest
+    {
+        ...test code...
+    }
+    ---
 
-------
-unittest
-{
-    ...test code...
-}
-------
+    $(P Individual tests are specified in the unit test using $(DDSUBLINK spec/expression, AssertExpression)s.
+    Unlike $(I AssertExpression)s used elsewhere, the assert is not assumed to hold, and upon assert
+    failure the program is still in a defined state.
+    )
 
     $(P There can be any number of unit test functions in a module,
     including within struct, union and class declarations.
     They are executed in lexical order.
-    The order in which the modules are called to run their unit tests is
-    implementation defined.
-    Stylistically, a unit test for a function should appear immediately
-    following it.
     )
 
-    $(P A compiler switch, such as $(DDSUBLINK dmd, switch-unittest, $(B -unittest))
-    for $(B dmd), will
-    cause the unittest test code to be compiled and incorporated into
-    the resulting executable. The unittest code gets run after
-    static initialization is run and before the $(D main())
-    function is called.
+    $(P Unit tests, when enabled, are run after all static initialization is
+    complete and before the $(D main()) function is called.
     )
 
-    $(P $(GLINK UnitTest)s must be grammatically correct even if $(B -unittest) is not
-    used.)
-
     $(P For example, given a class $(D Sum) that is used to add two values, a unit
     test can be given:)
 
-------
-class Sum
-{
-    int add(int x, int y) { return x + y; }
-
-    unittest
+    ---
+    class Sum
     {
-        Sum sum = new Sum;
-        assert(sum.add(3,4) == 7);
-        assert(sum.add(-2,0) == -2);
+        int add(int x, int y) { return x + y; }
+
+        unittest
+        {
+            Sum sum = new Sum;
+            assert(sum.add(3,4) == 7);
+            assert(sum.add(-2,0) == -2);
+        }
     }
-}
-------
+    ---
 
-$(H2 $(LNAME2 attributes_unittest, Attributed Unittests))
+    $(P When unit tests are enabled, the $(DDSUBLINK spec/version, PredefinedVersions, version identifier)
+    $(D unittest) is predefined.
+    )
 
-$(P A unittest may be attributed with any of the global function attributes.
-Such unittests are useful in verifying the given attribute(s) on a template
-function:)
 
-------
-void myFunc(T)(T[] data)
-{
-    if (data.length > 2)
-        data[0] = data[1];
-}
+$(H3 $(LNAME2 attributes_unittest, Attributed Unittests))
```

Comment:
Needs to be `H2`

Reply:
It would have taken me hours to figure that out based on that error message. Thanks!